### PR TITLE
Always run long CI

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Thanks to some speed optimizations, I think it's fairly safe to run the long tests on every branch now. Let's give it a shot.

This PR deletes a bunch of the code that was controlling when tests run.

**Testing steps**
New CI.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -448,31 +448,6 @@ jobs:
             echo 'Testing docs css'
             curl -f -i 'https://parl.ai/docs/_static/css/parlai_theme.css'
 
-  check_extra_tests:
```
Comment (EricMichaelSmith) [inline .circleci/config.yml:451]: Oh hmm are these tests duplicates of others, since they're being deleted?

File: codecov.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 coverage:
   status:
     project: off
-    patch: off
+    patch: on
```
Comment (EricMichaelSmith) [inline codecov.yml:4]: What does this control?

Comment (stephenroller) [reply codecov.yml:4]: I think it controls whether codecov makes the posts on PRs or not.

File: .circleci/config.yml (hunk 1 above)
Comment (stephenroller) [reply .circleci/config.yml:451]: This was the code that parsed the [long] bits.

**Pull request merged**